docs/source/examples/beam.py

```diff
@@ -47,6 +47,8 @@ def __init__(self, N=4, **kwargs):
         theta_eq[0] = (th[0] >= th_base)  # base boundary condition
         displ_eq = (w >= w.left + 0.5*dx*(th + th.left))
         displ_eq[0] = (w[0] >= w_base)
+        for constraint in [shear_eq, moment_eq, theta_eq, displ_eq]:
+            constraint.reset_varkeys()
```

Comment:
I mentioned a couple hours ago I didn't like that. :p Mostly because it makes setitem slow, and if users have updated their constraintset's varkeys by hand they won't expect those varkeys to disappear just because they used setitem.